packages/babel-compiler/babel-compiler.js

```diff
@@ -192,13 +197,37 @@ BCp.processOneFileForTarget = function (inputFile, source) {
                   tsx: hasTSXSupport,
                 },
               },
-              module: { type: 'commonjs' },
+              module: { type: 'es6' },
               minify: false,
               sourceMaps: true,
             });
 
+            let content = transformed.code;
+            // Perserve Meteor's specifics: reify modules, nested imports and top-level await support.
+            const result = reifyCompile(content, {
+              parse: reifyAcornParse,
+              generateLetDeclarations: false,
+              ast: false,
+              // Enforce reify options for proper compatibility (TODO export getReifyOptions)
+              // https://github.com/meteor/meteor/blob/devel/npm-packages/meteor-babel/options.js#L19
+              avoidModernSyntax: true,
+              enforceStrictMode: false,
+              dynamicImport: true,
+              ...features.topLevelAwait && { topLevelAwait: true },
+              ...features.compileForShell && { moduleAlias: 'module' },
+              ...(features.modernBrowsers ||
+                features.nodeMajorVersion >= 8) && {
+                avoidModernSyntax: false,
+                generateLetDeclarations: true,
+              },
```

Comment:
@zodern: Adding these options definitely improved the build time. Previously, the top-level await was triggering a Babel fallback instead of using SWC + Reify. Now it’s resolved, and the build times are consistent. 😁